enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,519 @@
+the density of the cluster (number of containers per worker node).
+On a fresh OpenShift cluster (with no user workload and only OpenShift own containers), it takes ~15 seconds to scan all containers.
+For a 8-worker node with 800 containers, assuming the containers are spread evenly, it would take around 20 seconds to scan the whole cluster. In practice, it will be a bit longer and capped by the scanning of the worker node with the most containers.
+
+The additional data in the `workloadRuntimeInfoContainer` will increase the size of the insights payload that is archived and sent to Red Hat backends.
+
+At most, the additional data adds __~200 uncompressed bytes per WorkloadContainerShape in the JSON payload__.
+
+This additional payload does not scale with the number of pods but with the number of `WorkloadContainerShape` which is difficult to estimate. There is an upper limit of 8000 pods
+in the Insights operator but that does not give a limit on the number of containers. It is also very likely that some of the pods are replicas and their containers would only count for 1 `WorkloadContainerShape` in their gathered data.
+
+#### UX Risk & Mitigation
+
+The additional runtime info added to the gathered data will be exposed to the user though the archives generated by the Insights operator. As the values of the field are hashed, the user will not be able to determine their corresponding content.
+
+Exposing these data to the user with a clean user interface (eg within console.redhat.com) is out of scope of this proposal.
+
+### Drawbacks
+
+If implemented, Red Hat will document the addition of this new collection. We  introduce the risk that some number of customers who would have or are sending telemetry data via the Insights operator will choose to turn it off because they are uncomfortable sharing this additional data. 
+
+The container scanner needs to be continually updated to detect new workloads  that Red Hat wants information about. 
+This can be done iteratively and will require an updated reference to the container scanner image in the Insights Operator to take into account the additional detections.
+
+As the container scanner is sanitizing the data extracted from the containers, it is not able to retrieve "raw" data (such as processes' name and command line) which limits its ability to detect emerging trends. The container scanner must provide fingerprints executables for workloads that Red Hat wants to monitor.
+
+## Design Details
+
+### Open Questions [optional]
+
+#### Container Scanner Image Availability
+
+The Insights Operator must be able to deploy the container scanner from an image that is available by default in the cluster image registry.
+
+How can we configure a _related image_ in the Insights Operator from its `ClusterVersion` What is the equivalent of a ClusterServiceVersion (CSV)’s `.spec.relatedImages` for OpenShift cluster operators?
```

Comment:
after discussion with the ART team, all is needed is to add this image to https://github.com/openshift/insights-operator/blob/master/manifests/image-references. 
I'll update the RFE